lib/mutant/integration.rb

```diff
@@ -4,38 +4,23 @@ module Mutant
   class Integration
     include AbstractType, Adamantium::Flat, Concord.new(:config)
 
-    REGISTRY = {}
-
     # Setup integration
     #
-    # @param [String] name
+    # Integrations are supposed to define a constant under
+    # Mutant::Integration named after tha capitalized +name*
+    # parameter.
     #
-    # @return [Integration]
-    def self.setup(name)
-      require "mutant/integration/#{name}"
-      lookup(name)
-    end
-
-    # Lookup integration for name
+    # This avoids having to maintain a mutable registry.
     #
-    # @param [String] name
+    # @param kernel [Kernel]
+    # @param name [String]
     #
-    # @return [Integration]
-    #   if found
-    def self.lookup(name)
-      REGISTRY.fetch(name)
+    # @return [Class<Integration>]
+    def self.setup(kernel, name)
```

Comment:
I do not use kwargs yet in this code base. Also this is an internal API right now a positional argument is fine for me, given this only takes 2 args.